platform/core/src/services/DisplaySetService/DisplaySetService.ts

```diff
@@ -236,34 +269,70 @@ export default class DisplaySetService extends PubSubService {
         );
 
         if (displaySets.length) {
-          this._addActiveDisplaySets(displaySets);
-        } else {
-          displaySets = handler.getDisplaySetsFromSeries(instances);
+          instances = filterInstances(instances, displaySets);
+          for (const ds of displaySets) {
+            const addedDs = ds.addInstances?.(instances, this);
+            if (addedDs) {
+              this.activeChanged = true;
+              instances = filterInstances(instances, [ds]);
+              this._addActiveDisplaySets([ds]);
+              this.setDisplaySetMetadataInvalidated(ds.displaySetInstanceUID);
+            }
+            if (!instances.length) return allDisplaySets;
+          }
+          if (instances.length === 0) {
```

Comment:
please be consistent you have `(!instances.length)` and `instances.length ===0` two lines apart from each other